### DISCUSSION TITLE
SOLID principles don't really apply to Rust programs

### DISCUSSION THREAD

**Discussion by oriongonza** [rust-unofficial/patterns#406] [General]
The obvious one here is the Liskov substitution principle which just doesn't make sense in Rust as there is no inheritance.
The Open Closed principle just doesn't make sense anymore.
The Interface Segregation principle is very debatable and shouldn't be stated as a fact.

I don't have a problem with the other two.

**Comment by simonsan** [comment]
Converted it to a discussion, as that fits better than an actual issue about content.

Not able to write a lot, currently, due to time constraints, so just leaving a link here for now: https://stackoverflow.com/questions/5577054/solid-for-functional-programming

**Reply by roberlamerma** [reply]
In spite it lacks more in-depth explanations (specially for Dependency Injection), I find this Quora post quite useful:

- https://www.quora.com/What-are-the-solid-principles-for-the-Rust-programming-language-and-software-design
